documentation/projects/proposals/trust_and_safety/content_report_moderation/20240122-implementation_plan_bulk_moderation_actions.md

```diff
@@ -0,0 +1,526 @@
+
+[See "ModerationDecisions List View" section for details](#moderationdecisions-list-view)
+
+- Add the list view
+- Add read-only detail view
+
+## Dependencies
+
+### Infrastructure
+
+<!-- Describe any infrastructure that will need to be provisioned or modified. In particular, identify associated potential cost changes. -->
+
+No changes needed.
+
+### Other projects or work
+
+<!-- Note any projects this plan is dependent on. -->
+
+This is heavily tied to the
+[related IP](https://docs.openverse.org/projects/proposals/trust_and_safety/content_report_moderation/20231208-implementation_plan_django_admin_moderator_access.html)
+and in fact changes some pieces of it slightly.
+
+## Alternatives
+
+<!-- Describe any alternatives considered and why they were not chosen or recommended. -->
+
+Several alternative ideas were explored as part of this work:
+
+### Creating a separate BulkModerationDecision model
+
+Many variations on creating additional models were considered and rejected
+because:
+
+- It resulted in additional complexity for ultimately very similar models
+- By having a single model instead we get many things for free, such as:
+  - bulk decisions are visible in the Report view, which is valuable context for
+    moderators
+  - individual decisions are visible in the ModerationDecision view
+  - the DeletedMedia and SensitiveMedia views allow filtering by id for any type
+    of moderation decision
+
+For example, one option was to create a separate ModerationDecision for each
+record in a bulk decision, with an optional foreign_id to the
+BulkModerationDecision parent object. This would result in potentially massive
+duplication of information, however, if large bulk decisions are made.
+
+I also considered having `provider` and `creator` as columns on a special type
+of ModerationDecision for all of a particular creator's records.
+
+- This adds complexity and even more ModerationDecision types; alternatively,
+  these would be nullable columns that are not always applicable (again, adding
+  complexity)
+- May cause confusion by implying that a record which does have these fields
+  should apply to absolutely all records by that creator, when this may not be
+  the case (for example, if some records by this creator have already been
+  deindexed/marked sensitive by previous decisions)
+- Ultimately this context should be included in the `explanation` field
+  regardless
+
+### Having a special form for creating ModerationDecisions that affect all records by a creator
+
+I considered including the ability to add a ModerationDecision directly, with a
+modified form that would take `creator` and `provider` rather than selecting the
+individual records to add.
+
+However, this duplicates some work, and makes it much more confusing to explain
+the various ways that a decision can be made. It is also already possible to do
+this sort of bulk action by filtering in the Media views on creator/provider.
```

Comment:
:+1: I like the thought process of "one obvious way to do things"! Very [zen](https://peps.python.org/pep-0020/).